apps/antalmanac/src/components/RightPane/CoursePane/SearchForm/DeptSearchBar/DeptSearchBar.tsx

```diff
@@ -1,175 +1,101 @@
+        }
 
-interface DeptSearchBarState {
-    value: Department;
-    favorites: Department[];
-}
+        if (data.every((x) => typeof x === 'object' && x !== null && 'deptValue' in x)) {
+            return data.map((x) => x.deptValue);
+        }
 
-class DeptSearchBar extends PureComponent<DeptSearchBarProps, DeptSearchBarState> {
-    updatedeptLabelAndGetFormData() {
-        RightPaneStore.updateFormValue('deptLabel', RightPaneStore.getUrlDeptLabel());
-        RightPaneStore.updateFormValue('deptValue', RightPaneStore.getUrlDeptValue());
-        return RightPaneStore.getFormData().deptLabel;
+        return [];
+    } catch (e) {
+        console.error('An error occurred:', e);
+        return [];
     }
+};
 
-    updatedeptValueAndGetFormData() {
-        RightPaneStore.updateFormValue('deptValue', RightPaneStore.getUrlDeptValue());
-        return RightPaneStore.getFormData().deptValue;
-    }
+export function DepartmentSearchBar() {
+    const [value, setValue] = useState(() => RightPaneStore.getFormData().deptValue);
+    const [favorites, setFavorites] = useState<typeof options>(() => parseLocalStorageFavorites());
 
-    getDeptValue() {
-        return RightPaneStore.getUrlDeptValue().trim()
-            ? this.updatedeptValueAndGetFormData()
-            : RightPaneStore.getFormData().deptValue;
-    }
+    const resetField = useCallback(() => {
+        setValue(() => RightPaneStore.getFormData().deptValue);
+    }, []);
 
-    getDeptLabel() {
-        return RightPaneStore.getUrlDeptLabel().trim()
-            ? this.updatedeptLabelAndGetFormData()
-            : RightPaneStore.getFormData().deptLabel;
-    }
+    const handleChange = useCallback(
+        (_: unknown, option: string | null) => {
+            const newValue = option ?? options[0]; // options[0] corresponds to `ALL`
 
-    constructor(props: DeptSearchBarProps) {
-        super(props);
+            setValue(newValue);
+            RightPaneStore.updateFormValue('deptValue', newValue);
 
-        let favorites: Department[] = [];
-        if (typeof Storage !== 'undefined') {
-            const locallyStoredFavorites = getLocalStorageFavorites();
-            favorites = locallyStoredFavorites != null ? JSON.parse(locallyStoredFavorites) : [];
-        }
-        this.state = {
-            value: {
-                deptValue: this.getDeptValue(),
-                deptLabel: this.getDeptLabel(),
-                isFavorite: false,
-            },
-            favorites: favorites,
-        };
-    }
+            const stateObj = { url: 'url' };
+            const url = new URL(window.location.href);
+            const urlParam = new URLSearchParams(url.search);
 
-    componentDidMount() {
-        RightPaneStore.on('formReset', this.resetField);
-    }
+            urlParam.delete('deptValue');
 
-    componentWillUnmount() {
-        RightPaneStore.removeListener('formReset', this.resetField);
-    }
+            if (newValue && newValue != 'ALL') {
```

Comment:
nit: redundant to check for truthyness
```suggestion
            if (newValue != 'ALL') {
```